auditbeat/module/auditd/audit_linux.go

```diff
@@ -204,9 +207,107 @@ func (ms *MetricSet) Run(reporter mb.PushReporterV2) {
 			}
 		}()
 	}
+
+	if ms.config.Reload.Enabled {
+		ruleReloader := newRuleReloader(ms.config.RuleFiles, ms.config.Reload)
+
+		go ruleReloader.Run(ms, reporter)
+	}
+
 	wg.Wait()
 }
 
+type reloader struct {
+	reload reload
+	pathes []string
+	done   chan struct{}
+	wg     sync.WaitGroup
+}
+
+func newRuleReloader(pathes []string, reload reload) *reloader {
+
+	newPathes := []string{}
+
+	for _, path := range pathes {
+		if !filepath.IsAbs(path) {
+			path = paths.Resolve(paths.Config, path)
+		}
+		newPathes = append(newPathes, path)
+	}
+
+	return &reloader{
+		reload: reload,
+		pathes: newPathes,
+		done:   make(chan struct{}),
+	}
+}
+
+// Run runs the reloader
+func (rl *reloader) Run(ms *MetricSet, reporter mb.PushReporterV2) {
+	ms.log.Info("Auditd Rule reloader started")
+
+	rl.wg.Add(1)
+	defer rl.wg.Done()
+
+	gws := []*cfgfile.GlobWatcher{}
+	for _, path := range rl.pathes {
+		gws = append(gws, cfgfile.NewGlobWatcher(path))
+	}
+
+	// If reloading is disable, config files should be loaded immediately
+	if !rl.reload.Enabled {
+		rl.reload.Period = 0
+	}
+
+	for {
+		select {
+		case <-rl.done:
+			return
+
+		case <-time.After(rl.reload.Period):
+			ms.log.Debugf("Scan for new Auditd Rule files")
+
+			for _, gw := range gws {
+				_, updated, err := gw.Scan()
+				ms.log.Info("rule files updated:", updated)
+
+				if err != nil {
+					// In most cases of error, updated == false, so will continue
+					// to next iteration below
+					ms.log.Errorw("Error fetching new Auditd Rule files: %v", err)
+				}
+
+				// no file changes
+				if !updated {
+					continue
+				}
+
+				// Reload all config objects
+				if err := ms.config.reloadRules(); err != nil {
+					reporter.Error(err)
+					ms.log.Errorw("Failure reload audit rules from rule file", "error", err)
+					return
```

Comment:
Should the reloader be stopping on error? I imaging that when new errors I deployed failed to load, I will deploy more changes to replace the bad ones.